tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+      expect(ws5.options).toEqual options5
+
+
+    it 'should fire events per each instance', ->
+      emitter1 = false
+      ws1 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      ws1.on 'test', ->
+        emitter1 = true
+      
+      ws1.emit 'test'
+
+      emitter2 = false
+      ws2 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      ws2.on 'test', ->
+        emitter2 = true
+
+      ws2.emit 'test'
+
+      expect(emitter1).toBeTruthy()
+      expect(emitter2).toBeTruthy()
+
+  describe 'connect', ->
+
+    it 'should initialize socket when try to connect with different domain and token', ->
+      ws1 = new WebSocketTransport('wss://r5crcc.api.really.io','ibj88w5aye', options)
+      ws1.connect()
+      socket1 = ws1.socket
+      expect(socket1).toBeDefined()
+      ws2 = new WebSocketTransport('wss://r5crbb.api.really.io','ibj88w5aye', options)
+      ws2.connect()
+      socket2 = ws2.socket
+      expect(socket2).not.toBe(socket1)
+      ws1.disconnect()
+      ws2.disconnect()
+
+    it 'should trigger error event when server is blocked/not found', (done) ->
+      ws = new WebSocketTransport('wss://WRONG_ID.really.com','ibj88w5aye', options)
+      connected = true
+      ws.connect()
+      ws.on 'error', () ->
+        connected = false
+
+      setTimeout (->
+        expect(connected).toBeFalsy()
+        ws.disconnect()
+        done()
+      ), 2000
+
+    it 'should send first message', (done) ->
+      ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5aye', options)
+      ws.connect()
+      message = {tag: 1, 'cmd': 'init', accessToken: 'ibj88w5aye'}
+      ws.on 'message', (msg) ->
+        expect(message).toEqual msg
+        ws.disconnect()
+        done()
+
+    it 'should trigger initialized event with user data, after calling success callback', (done) ->
+      ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5aye', options)
+      ws.connect()
+      ws.on 'initialized', (data) ->
+        expect(ws.initialized).toBeTruthy()
+        ws.disconnect()
+        done()
+
+    
+    it 'should return a promise', ->
+      ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+      promise = ws.connect()
+      expect(typeof promise.then is 'function').toBeTruthy()
+      ws.disconnect()
+
+    describe 'connection open', ->
+      it 'should initialize reconnectiong attempts number to zero', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        ws.connect()
+        ws.socket.addEventListener 'open', () ->
+          expect(ws.attempts).toEqual 0
+          ws.disconnect()
+          done()
+
+      it 'should reslove a promise when connection open', (done) ->
+        ws = new WebSocketTransport(CONFIG.REALLY_DOMAIN, 'ibj88w5ake', options)
+        promise = ws.connect()
+        ws.socket.addEventListener 'open', () ->
+          promise.done( (data) ->
```

Comment:
how can I spy on `done`? it will not be called in the code ... I just called it in the test to check if the promise is resolved.. either I can do this `expect(typeof promise.done is 'function').toBeTruthy()` but for me it doesn't need to be changed, what do think?